src/components/chart/bar-chart.js

```diff
@@ -0,0 +1,167 @@
+                        },
+                    },
+                },
+                series: {
+                    stacking: useStackedLayout ? 'normal' : null,
+                },
+            },
+            xAxis: {
+                // Update the category label colours for bar charts
+                labels: {
+                    style: {
+                        color: this.constants.categoryLabelColor,
+                    },
+                },
+                // remove the tick marks for bar charts
+                tickWidth: 0,
+                tickLength: 0,
+                tickColor: 'transparent',
+                title: { align: 'high', textAlign: 'middle', reserveSpace: false, rotation: 0, y: -25 },
+            },
+            yAxis: {
+                title: {
+                    // Override the y Axis title settings for bar charts where the y axis is horizontal
+                    textAlign: 'right',
+                    offset: undefined,
+                    y: 0,
+                    reserveSpace: true,
+                },
+            },
+        };
+    };
+
+    hideDataLabels = (config) => {
+        config.series.forEach((series) => {
+            series.dataLabels = {
+                enabled: false,
+            };
+        });
+    };
+
+    // Updates the config to move the data labels inside the bars, but only if the bar is wide enough
+    // This may also need to run when the chart is resized
+    postLoadDataLabels = (currentChart) => {
+        const insideOptions = {
+            dataLabels: this.getBarChartLabelsInsideOptions(),
+        };
+        const outsideOptions = {
+            dataLabels: this.getBarChartLabelsOutsideOptions(),
+        };
+
+        currentChart.series.forEach((series) => {
+            const points = series.data;
+            points.forEach((point) => {
+                // Get the actual width of the data label
+                const labelWidth = point.dataLabel && point.dataLabel.getBBox().width;
+                // Move the data labels inside the bar if the bar is wider than the label plus some padding
+                if (point.shapeArgs.height > labelWidth + 20) {
+                    point.update(insideOptions, false);
+                } else {
+                    point.update(outsideOptions, false);
+                }
+            });
+        });
+
+        currentChart.redraw();
+    };
+
+    getBarChartLabelsInsideOptions = () => ({
+        inside: true,
+        align: 'right',
+        verticalAlign: 'middle',
+        style: {
+            color: 'white',
+            fontWeight: 'bold',
+        },
+    });
+
+    getBarChartLabelsOutsideOptions = () => ({
+        inside: false,
+        align: undefined,
+        verticalAlign: undefined,
+        style: {
+            textOutline: 'none',
+            // there is no semibold font weight available in the design system fonts, so we use 700 instead
+            fontWeight: '700',
```

Comment:
Our bold font weight is 700

Reply:
Oh I think I got confused by the wording of the comment, it might be good to update that to be a bit more clear. It seems to read to me that you have decided to use 700 because there is no semi bold where 700 weight would be the ideal semi bold weight

Reply:
updated